haicrypt/cryspr.c

```diff
@@ -122,10 +124,14 @@ int crysprStub_KmPbkdf2(
 
 static int crysprFallback_KmSetKey(CRYSPR_cb *cryspr_cb, bool bWrap, const unsigned char *kek, size_t kek_len)
 {
-	CRYSPR_AESCTX *aes_kek = &cryspr_cb->aes_kek;
-
-    if (cryspr_cb->cryspr->aes_set_key(bWrap, kek, kek_len, aes_kek)) {
-        HCRYPT_LOG(LOG_ERR, "AES_set_%s_key(kek) failed\n", bWrap? "encrypt": "decrypt");
+#ifdef CRYSPR2
+    CRYSPR_AESCTX *aes_kek =  cryspr_cb->aes_kek;
+#else /*CRYSPR2*/
+    CRYSPR_AESCTX *aes_kek = &cryspr_cb->aes_kek;
+#endif /*CRYSPR2*/
```

Comment:
Adding a getter function (e.g. `cryspr_cb->getKEK()` and a similar one for SEK) would localize such `ifdef`s in one place instead of having them everywhere.

Reply:
Implemented with a macro